DXRCore/DeusEx/Classes/DXRActorsBase.uc

```diff
@@ -86,12 +86,40 @@ static function bool IsRobot(class<Actor> a)
     return ClassIsChildOf(a, class'Robot');
 }
 
-static function bool IsCritter(Actor a)
+static function bool IsCombatRobot(class<Actor> a)
 {
-    if( #var(prefix)CleanerBot(a) != None ) return true;
-    if( #var(prefix)MedicalBot(a) != None || #var(prefix)RepairBot(a) != None || Merchant(a) != None ) return true;
-    if( #var(prefix)Animal(a) == None ) return false;
-    return #var(prefix)Doberman(a) == None && #var(prefix)Gray(a) == None && #var(prefix)Greasel(a) == None && #var(prefix)Karkian(a) == None;
+    return
+        IsRobot(a) &&
+        !ClassIsChildOf(a, class'#var(prefix)MedicalBot') &&
+        !ClassIsChildOf(a, class'#var(prefix)RepairBot') &&
+        !ClassIsChildOf(a, class'#var(prefix)CleanerBot');
+}
+
+// Returns True if the class is an animal that will never fight
+static function bool IsCritter(class<Actor> a)
+{
+    return
+        ClassIsChildOf(a, class'#var(prefix)Animal') &&
+        !ClassIsChildOf(a, class'#var(prefix)Dog') &&
+        !ClassIsChildOf(a, class'#var(prefix)Gray') &&
+        !ClassIsChildOf(a, class'#var(prefix)Greasel') &&
+        !ClassIsChildOf(a, class'#var(prefix)Karkian');
+}
+
+static function bool IsCombatAnimal(class<Actor> a)
+{
+    return ClassIsChildOf(a, class'#var(prefix)Animal') && !IsCritter(a);
+}
+
+// Returns True if the class will ever fight, even if some instances won't
+static function bool IsCombatPawn(class<Actor> a)
+{
+    return IsHuman(a) || IsCombatRobot(a) || IsCombatAnimal(a);
+}
+
+static function bool IsSwappable(class<Actor> a)
+{
+    return IsCombatPawn(a) && !ClassIsChildOf(a, class'Merchant');
```

Comment:
this should maybe be checking for bImportant or bIsSecretGoal or whatever?
```suggestion
    return IsCombatPawn(a) && !a.bImportant && !a.bIsSecretGoal;
```

Reply:
The code that IsSwappable replaced (the old, misnamed IsCritter function) only checked the class, rather than anything about that specific pawn. I've renamed this function from IsSwappable to IsRelevantPawn to make its purpose more clear.